### PR TITLE
Handle relative paths more correctly

### PR DESCRIPTION
**Pull request by epsilon-phase**

### CODE REVIEW

File: pkg/fs/apkfs.go [hunk 1]
```diff
@@ -235,8 +235,17 @@ func (a *APKFS) Stat(path string) (fs.FileInfo, error) {
...
 }
-
+func correctApkFSPath(path string) string {
+	if path == "." {
+		path = "/"
+	}
+	if path[:2] == "./" {
+		path = path[1:]
+	}
+	return filepath.Clean(path)
```
Comment (Elizafox) [inline pkg/fs/apkfs.go:245]: should probably clean before you do those checks

File: pkg/fs/apkfs.go [hunk 2]
```diff
@@ -235,8 +235,17 @@ func (a *APKFS) Stat(path string) (fs.FileInfo, error) {
 	}
 	return &apkFSFileInfo{file: file, name: file.name[strings.LastIndex(file.name, "/"):]}, nil
 }
-
+func correctApkFSPath(path string) string {
+	if path == "." {
+		path = "/"
+	}
+	if path[:2] == "./" {
```
Comment (Elizafox) [inline pkg/fs/apkfs.go:242]: You need to verify the length of path before taking a slice of this length or you will cause a panic.

**Review by Elizafox — CHANGES_REQUESTED** [review]

File: pkg/fs/apkfs.go (hunk 1 above)
Comment (epsilon-phase) [reply pkg/fs/apkfs.go]: Switching the order breaks it so,,, I'm thinking I'll just leave it as is

File: pkg/fs/apkfs.go [hunk 3]
```diff
@@ -235,8 +235,18 @@ func (a *APKFS) Stat(path string) (fs.FileInfo, error) {
 	}
 	return &apkFSFileInfo{file: file, name: file.name[strings.LastIndex(file.name, "/"):]}, nil
 }
-
+func correctApkFSPath(path string) string {
+	if path == "." {
+		path = "/"
+	}
+	if len(path) > 3 && path[:2] == "./" {
```
Comment (Elizafox) [inline pkg/fs/apkfs.go:242]:
```suggestion
	if len(path) >= 2 && path[:2] == "./" {
```

File: pkg/fs/apkfs.go [hunk 4]
```diff
@@ -235,8 +235,18 @@ func (a *APKFS) Stat(path string) (fs.FileInfo, error) {
 	}
 	return &apkFSFileInfo{file: file, name: file.name[strings.LastIndex(file.name, "/"):]}, nil
 }
-
+func correctApkFSPath(path string) string {
+	if path == "." {
```
Comment (Elizafox) [inline pkg/fs/apkfs.go:239]:
```suggestion
	path = filepath.Clean(path)

	if path == "." {
```

Move the cleaning before checking.

File: pkg/fs/apkfs.go [hunk 5]
```diff
@@ -235,8 +235,18 @@ func (a *APKFS) Stat(path string) (fs.FileInfo, error) {
...
+		path = "/"
+	}
+	if len(path) > 3 && path[:2] == "./" {
+		path = path[1:]
+	}
+	path = filepath.Clean(path)
```
Comment (Elizafox) [inline pkg/fs/apkfs.go:245]:
```suggestion
```

Move that.

**Review by Elizafox — CHANGES_REQUESTED** [review]

File: pkg/fs/apkfs.go (hunk 5 above)
Comment (epsilon-phase) [reply pkg/fs/apkfs.go:245]: Moving it before the other checks breaks the path resolution

**Review by Elizafox — APPROVED** [review]

**Pull request merged**